src/java.base/share/classes/jdk/internal/util/HexDigits.java

```diff
@@ -194,4 +194,37 @@ public static int stringSize(long value) {
         return value == 0 ? 1 :
                 67 - Long.numberOfLeadingZeros(value) >> 2;
     }
+
+    /**
+     * Extract the least significant 8 bytes from the input integer i, convert each byte into its corresponding 2-digit
+     * hexadecimal representation, concatenate these hexadecimal strings into one continuous string, and then interpret
+     * this string as a hexadecimal number to form and return a long value.
+     */
+    public static long hex8(long i) {
+        long x = Long.expand(i, 0x0F0F_0F0F_0F0F_0F0FL);
```

Comment:
x86 should use  pepd - but aarch64?

Reply:
Seems there is no good way to do so on aarch; raw shifts and going through VPU both seem to be slower.